crates/payload/builder/src/service.rs

```diff
@@ -200,7 +208,10 @@ impl<Gen, St, Engine> PayloadBuilderService<Gen, St, Engine>
 where
     Engine: EngineTypes,
     Gen: PayloadJobGenerator,
-    Gen::Job: PayloadJob<PayloadAttributes = Engine::PayloadBuilderAttributes>,
+    Gen::Job: PayloadJob<
+        PayloadAttributes = Engine::PayloadBuilderAttributes,
+        BuiltPayload = Engine::BuiltPayload,
```

Comment:
I think we want:

```rust
<Gen::Job as PayloadJob>: Into<Engine::BuiltPayload>
```

this way the service can support any generator that creates jobs with a Payload that can be converted to the configured Engine::BuiltPayload.

I realize that this breaks the Arc wrapper, so I think we should remove the Arc from the Job trait function signature.

Reply:
alright, I had to get rid of pretty much all of the Arc, might be possible to make it work again with `best_payload`, but the bounds are now `Into<Engine::BuiltPayload>`